### PR TITLE
fix: support outbound sync in ui

### PR DESCRIPTION
**Pull request by asdfryan**
Screenshot 2023-02-16 at 1 04 02 PM

### CODE REVIEW

File: packages/nextjs/src/elements/FieldMapping/FieldMapping.tsx [hunk 1]
```diff
@@ -118,7 +110,16 @@ const FieldCollection = ({ appearance, syncConfig, sync }: FieldCollectionProps)
...
 
-  const { upsertKey } = (syncConfig.destination as PostgresDestination).config;
+  const getUpsertKey = (syncConfig: SyncConfig): string | undefined => {
+    if (syncConfig.type === 'outbound') {
+      return syncConfig.destination.upsertKey;
+    }
+    if (syncConfig.destination.type === 'postgres') {
+      return syncConfig.destination.config.upsertKey;
+    }
+  };
```
Comment (khennes) [inline packages/nextjs/src/elements/FieldMapping/FieldMapping.tsx:120]: nit: this function could be defined outside the component (maybe in a lib or utils file instead?)

**Review by khennes — COMMENTED** [review]
lg, just need to fix:
```
Error: @supaglue/nextjs:build: src/elements/FieldMapping/FieldMapping.tsx(26,34): error TS2304: Cannot find name 'PostgresDestination'.
Error: @supaglue/nextjs:build: src/elements/FieldMapping/FieldMapping.tsx(94,31): error TS2304: Cannot find name 'PostgresDestination'.
```

File: packages/types/src/util.ts [hunk 2]
```diff
@@ -13,3 +17,26 @@ const getDefaultObject = (salesforceObjectConfig: SalesforceObjectConfig): Sales
...
+  }
+  return syncConfig.destination.schema;
+};
+
+export const customPropertiesEnabled = (syncConfig: SyncConfig): boolean => {
+  if (syncConfig.type === 'outbound') {
+    return !!syncConfig.source.config.customPropertiesColumn;
+  }
+  return syncConfig.destination.type === 'postgres' && !!syncConfig.destination.config.customPropertiesColumn;
+};
```
Comment (khennes) [inline packages/types/src/util.ts:42]: Heads up, custom properties are not supported yet for any syncs except for inbound -> postgres syncs.

Comment (asdfryan) [reply packages/types/src/util.ts:42]: yeah that's fine -- our outbound config doesn't have it anyway.

**Review by khennes — APPROVED** [review]

**Pull request merged**